libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -608,7 +608,7 @@ class MapboxNavigation(
 
     private fun createInternalOffRouteObserver() = object : OffRouteObserver {
         override fun onOffRouteStateChanged(offRoute: Boolean) {
-            if (offRoute) {
+            if (offRoute && tripSession.route != null) {
```

Comment:
`offRoute && tripSession.route == null` is a bug, we shouldn't need to keep this if-check if the rest of the logic works correctly. That's why we need to synchronize the route setter and navigator status getter (and all of the observer updates). Unless I'm missing something?

Reply:
What I'm trying to say is that the issue described in https://github.com/mapbox/mapbox-navigation-android/issues/3383 can still happen if I'm not mistaken:
```
1. getStatus called on a background thread and reports off-route
2. developer calls setRoutes(empty) on the main thread
3. status from 1. is switched to the main thread and delivered to the developer
```

Reply:
Even if `offroute` is guarded, the rest of the progress updates are still delivered, even though the developer has cleared the route already.